### PR TITLE
Improve safe UI

### PR DESCRIPTION
**Pull request by valentinludu**
### <samp>🤖 Generated by Copilot at 11627c1</samp>

Improved the UI and functionality of the multi-payment feature for bounties, allowing users to manage their wallets more easily and securely. Refactored some components and hooks to use better types and data fetching methods. Added a new feature to the `useSettingsDialog` hook to scroll to a specific section. Modified the `CharmClient` class and the `/api/profile/gnosis-safes/[id]` endpoint to support wallet visibility updates.

### WHY
[Task item](https://app.charmverse.io/charmverse/page-9962561669746739)

### CODE REVIEW

**Comment by motechFR** [comment]
@valentinludu I cleaned up a bug in the bounties so that it shows the error message for the current network when no safes are found.

I found this bug, where the menu offers me multiple Gnosis Safes, even though the bounties I am seeing are restricted to current chainID.

Can confirm, that hidden bounties do work though

Screenshot 2023-05-11 at 19 01 00

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx [hunk 1]
```diff
@@ -231,18 +231,20 @@ export function BountyPaymentButton({
...
-              label={safeDataRecord[safeInfo.address]?.name ?? shortenHex(safeInfo.address)}
-              onClick={() => {
-                handleClose();
-              }}
-              onError={onError}
-              safeInfo={safeInfo}
-            />
-          ))}
+          {safesData
+            ?.filter((s) => !s.isHidden)
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:235]: Relating to bug I raised, we might want to add a condition on the filter that safe.chainId === bounty.chainId

File: components/integrations/components/MultiSigList.tsx [hunk 2]
```diff
@@ -185,16 +153,13 @@ function SafeRow({ safe, updateWallets }: { safe: Safe; updateWallets: () => voi
...
-          onConfirm={() => deleteSafe(safe)}
-          onClose={deleteConfirmation.close}
-          open={deleteConfirmation.isOpen}
-        />
+        <IconButton
+          disabled={isLoadingUpdateSafeWallet}
+          size='small'
+          onClick={() => updateSafeWallet({ id: safe.id, isHidden: !safe.isHidden })}
+        >
+          {safe.isHidden ? <VisibilityOffOutlinedIcon /> : <VisibilityOutlinedIcon />}
```
Comment (motechFR) [inline components/integrations/components/MultiSigList.tsx:161]: Suggestion: Lower opacity for a hidden safe

Comment (valentinludu) [reply components/integrations/components/MultiSigList.tsx:161]: Good idea.
Done!

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx (hunk 1 above)
Comment (valentinludu) [reply components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx]: Done

**Pull request merged**